File: wasm/examples/filter.rs
#[derive(Clone)]
#[no_mangle]
pub struct Event {
    id: String,
    pubkey: String,
    created_at: i64, // unix timestamp in seconds
    kind: i32,
    tags: Vec<Vec<String>>,
    content: String,
    sig: String,
}

#[no_mangle]
pub fn isValidEvent(event: Event) -> bool {
    event.kind == 1
}

File: src/pages/noscript/test/filter.ts
import { initSync, is_valid_event } from "../filter-binding";
import { Event } from 'core/nostr/Event';

export const testFn = async (byteCode: ArrayBuffer) => {
  initSync(byteCode);
  const event: Event = {
    id: "aaa",
    pubkey: "bbb",
    sig: "ccc",
    content: "",
    kind: 1,
    created_at: 1000,
    tags: []
  }
  console.log("is_valid_event(event): ", is_valid_event, is_valid_event(event));
}
